centralized-sampling-tests/integration-tests/src/main/java/com.amazon.tests/CentralizedSamplingIntegrationTests.java

```diff
@@ -0,0 +1,313 @@
+package com.amazon.tests;
+
+import java.io.*;
+import java.lang.Math;
+import java.util.concurrent.TimeUnit;
+import okhttp3.*;
+import org.springframework.context.annotation.Bean;
+
+/** File that runs all of the integrations tests and creates/deletes sample rules */
+public class CentralizedSamplingIntegrationTests {
+
+  /**
+   * Set up new httpClient
+   *
+   * @return OkhttpClient
+   */
+  @Bean
+  public static Call.Factory httpClient() {
+    return new OkHttpClient();
+  }
+
+  /**
+   * Main function that runs the three tests for Centralized Sampling SampleRulesTests - tests rules
+   * filter and sample the correct number of targets ReservoirTests - tests that reservoir works
+   * correctly with Sampling Rules PriorityTests - test Priority of sampling rules that higher
+   * priority rules get sampled first
+   *
+   * @param args
+   * @throws IOException - throws if calls to xray backend fail
+   * @throws InterruptedException - throws if tests fail
+   */
+  public static void main(String[] args) throws IOException, InterruptedException {
+    sampleRulesTests();
+    reservoirTests();
+    priorityTests();
+  }
+
+  /**
+   * Function to
+   *
+   * @param testCase - testCase to make call with, determines headers, method, and endpoint, etc.
+   * @param sampleRule - sampleRule that is currently active
+   * @return true if call is successful and has expected sampling rate, false else
+   */
+  public static boolean makeCalls(testCases testCase, SampleRule sampleRule) throws IOException {
+    RequestBody reqbody = null;
+    String stringResp = "";
+    if (testCase.method.equals("POST")) {
+      reqbody = RequestBody.create(null, new byte[0]);
```

Comment:
If method is "GET", reqbody should be null?

Reply:
yes GET requests cannot have requestBody

Reply:
also weird because POSTs are required to have a reqbody even though I am not using it hence the empty reqbody